src/main.rs

```diff
@@ -128,11 +130,25 @@ fn generate(parent: Option<&Email>) -> Email {
             hdrs.insert(String::from("Subject"), new_subj.to_string());
         }
 
-        // TODO: parse parent date and move forwards.
-        let date: chrono::DateTime<Utc> = DateTime(EN).fake();
-        hdrs.insert(String::from("Date"), date.to_rfc2822());
+        let parent_date_hdr = m.headers.get("Date");
+        if (parent_date_hdr).is_some() {
+            let parent_date: chrono::DateTime<Utc> =
+                chrono::DateTime::parse_from_rfc2822(parent_date_hdr.unwrap())
+                    .unwrap()
+                    .into();
+            let date: chrono::DateTime<Utc> = DateTimeAfter(EN, parent_date).fake();
+            hdrs.insert(String::from("Date"), date.to_rfc2822());
+        } else {
+            let date: chrono::DateTime<Utc> = DateTime(EN).fake();
+            hdrs.insert(String::from("Date"), date.to_rfc2822());
+        }
```

Comment:
There are a couple ways to make this a little more convenient. In this case, since you care both about the `Some(foo)` case and the `None` case, I'd suggest `match`.

Another potential thing to note is that `if`/`else` and `match` can be used as the right-hand side of assignments, so you can assign them to a value and leave shared logic outside. (Note the absence of a semicolon at the end of the last lines of each branch.)

```suggestion
        let date: chrono::DateTime<Utc> = match parent_date_hdr {
            Some(hdr) => {
                let parent_date: chrono::DateTime<Utc> =
                    chrono::DateTime::parse_from_rfc2822(hdr)
                        .expect("Couldn't parse date in header")
                        .into();
                        
            DateTimeAfter(EN, parent_date).fake()
        } else {
            DateTime(EN).fake()
        };
        
        hdrs.insert(String::from("Date"), date.to_rfc2822());
```